addon/lib/field.js

```diff
@@ -407,6 +444,39 @@ export default Base.extend({
     }
   ),
 
+  /**
+   * Fields that are referenced in the `calcExpression` JEXL expression
+   *
+   * If the value or hidden state of any of these fields change, the JEXL
+   * expression needs to be re-evaluated.
+   *
+   * @property {Field[]} calculatedDependencies
+   * @accessor
+   */
+  calculatedDependencies: computed(
+    "document.{jexl,fields.[]}",
+    "question.{isCalculated,calcExpression}",
+    function () {
+      if (!this.question.isCalculated) {
+        return [];
+      }
+
+      return getDependenciesFromJexl(
+        this.document.jexl,
+        this.question.calcExpression
+      ).map((slug) => {
+        const f = this.document.findField(slug);
+
+        assert(
+          `Field for question \`${slug}\` was not found in this document. Please check the \`calcExpression\` jexl expression: \`${this.question.calcExpression}\`.`,
+          f
+        );
+
+        return f;
+      });
+    }
+  ),
```

Comment:
Did you already check if this has a noticable performance impact in larger forms?

Reply:
Not sure what you'd expect TBH. If you don't add like 20 fields with formulas using 20 different fields I'm not sure how this would impact the performance

Reply:
I didn't think too deeply about this, I just remember that we (or rather you :wink) optimized this code a bit in the past and computed properties can sometimes cause an overhead. I just wanted to bring this up, we can also check the performance side of things after the PR landed.

Reply:
It won't be a problem. There will be very few fields with content in this. Also, I optimized that dependency logic to exactly this (computed properties) because it the most performant. This will be ok.

Reply:
Alright :+1: